### PR TITLE
Update k8s docs for Vault as a Secrets Backend

### PR DESCRIPTION
**Pull request by jmurret**
This all of the current Vault changes except for the WAN Federation guide.

The main goal of this PR was to introduce adding support for bootstrap token, replication token, snapshot agent config, and enterprise license.

The other goal was to try to break up the various commands in to some further buckets. Integration requires a large run list of commands that is a little hard to grok everything that is occurring and why for those not intimately familiar with it.  I broke this documentation down into two sections:
- Systems Integration - what needs to be turned and configured so that the systems are aware of each other and ready to store and read data
- Data Integration - what needs to be done to store data and configure authorization so Consul on Kubernetes can read a specific secret.

I plan to add Wan Federation documentation as an end to end example at the same level as Systems Integration and Data Integration in the below nav tree.  Also WAN Federation documentation will use the same general steps structure that is used in the Systems Integration and Data Integration Overviews.

Screen Shot 2022-04-04 at 4 24 26 PM

### CODE REVIEW

File: website/content/docs/k8s/installation/vault/index.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/index.mdx:14]:
```suggestion
The following secrets can be stored in Vault KV secrets engine, which is meant to handle arbitrary secrets:
```

File: website/content/docs/k8s/installation/vault/systems-integration.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/systems-integration.mdx:25]:
```suggestion
- Enterprise license ([`global.enterpriseLicense`]())
```

File: website/content/docs/k8s/installation/vault/systems-integration.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/systems-integration.mdx:43]:
```suggestion
## Set up per Consul Datacenter
```

Datacenter is a proper noun for Consul representing a cluster deployed within a region.

File: website/content/docs/k8s/installation/vault/systems-integration.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/systems-integration.mdx:87]:
```suggestion
  externalVaultAddr: ${VAULT_ADDR}
```

File: website/content/docs/k8s/installation/vault/systems-integration.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/systems-integration.mdx:130]:
```suggestion
- Enterprise license ([`global.enterpriseLicense`]())
```

File: website/content/docs/k8s/installation/vault/data-integration/index.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/index.mdx:28]:
```suggestion
- Set up per Consul Datacenter
```

File: website/content/docs/k8s/installation/vault/data-integration/server-tls.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/server-tls.mdx:47]:
```suggestion
```


Perhaps we can remove this from the list but still mention it below in the case of posterity to previous setup instructions.

Comment (david-yu) [reply website/content/docs/k8s/installation/vault/data-integration/server-tls.mdx:47]: I found it a looking a little odd that this was striked out.

File: website/content/docs/k8s/installation/vault/data-integration/connect-ca.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/connect-ca.mdx:25]:
```suggestion
```

Same comment as https://github.com/hashicorp/consul/pull/12691/files#r842222392

File: website/content/docs/k8s/installation/vault/data-integration/gossip.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/gossip.mdx:21]:
```suggestion
- Set up per Consul datacenter
```

File: website/content/docs/k8s/installation/vault/data-integration/snapshot-agent-config.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/snapshot-agent-config.mdx:21]:
```suggestion
- Set up per Consul datacenter
```

File: website/content/docs/k8s/installation/vault/data-integration/bootstrap-token.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/bootstrap-token.mdx:21]:
```suggestion
- Set up per Consul datacenter
```

File: website/content/docs/k8s/installation/vault/data-integration/connect-ca.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/connect-ca.mdx:27]:
```suggestion
- Set up per Consul datacenter
```

File: website/content/docs/k8s/installation/vault/data-integration/enterprise-license.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/enterprise-license.mdx:21]:
```suggestion
- Set up per Consul datacenter
```

File: website/content/docs/k8s/installation/vault/data-integration/replication-token.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/replication-token.mdx:76]:
```suggestion
### Configure the Vault Kubernetes auth role in the Consul on Kubernetes helm chart
```

File: website/content/docs/k8s/installation/vault/data-integration/replication-token.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/replication-token.mdx:55]:
```suggestion
## Set up per Consul datacenter
```

File: website/content/docs/k8s/installation/vault/data-integration/server-tls.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/data-integration/server-tls.mdx:47]:
```suggestion
```

File: website/content/docs/k8s/installation/vault/systems-integration.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/systems-integration.mdx:12]:
```suggestion
## One time set up on Vault
```

File: website/content/docs/k8s/installation/vault/systems-integration.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/systems-integration.mdx:15]:
```suggestion
## Set up per Consul datacenter
```

File: website/content/docs/k8s/installation/vault/systems-integration.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/vault/systems-integration.mdx:122]:
```suggestion
    kubernetes_host="${https://$KUBE_API_URL}:443" \
```

**Review by david-yu — APPROVED** [review]
This is good to go from my side. I'll need to revisit again in the future to perhaps refactor the `helm template` commands to set env variables for use in the `vault write policy` commands. But it looks good so far! Thanks for your work on this, this format looks much more digestible for someone starting off with this feature.

**Review by ishustava — APPROVED** [review]
Love the reorganization of the Vault concepts!!

I'm leaving some edits and questions but won't block the merge. If I find anything else, I'll submit a separate PR to fix because I probably won't have time to review in more detail.

File: website/content/docs/k8s/installation/vault/systems-integration.mdx
Comment (jmurret) [reply website/content/docs/k8s/installation/vault/systems-integration.mdx:98]: No. This is in the existing docs here: https://www.consul.io/docs/k8s/installation/vault#vault-helm-config

Should we keep this?

Comment (ishustava) [reply website/content/docs/k8s/installation/vault/systems-integration.mdx:98]: I think it's ok to keep it, although maybe we can fix the helm value:
```suggestion
A minimal valid installation of Vault Kubernetes must include the Agent Injector which is utilized for accessing secrets from Vault. Vault servers could be deployed external to Vault on Kubernetes with the [`injector.externalVaultAddr`](https://www.vaultproject.io/docs/platform/k8s/helm/configuration#externalvaultaddr) value in the Vault Helm Configuration.
```

File: website/content/docs/k8s/installation/vault/systems-integration.mdx
Comment (jmurret) [reply website/content/docs/k8s/installation/vault/systems-integration.mdx:35]: I just created this page and set up all of the appropriate links in the various pages and nav.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/630282.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit fd08b6aaf62b0a915a45c6c4828fd1d532b6d26b onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit fd08b6aaf62b0a915a45c6c4828fd1d532b6d26b onto `release/1.12.x` succeeded!

**Pull request merged**
